leveldb/storage/storage.go

```diff
@@ -177,3 +178,121 @@ type Storage interface {
 	// called after the storage has been closed.
 	Close() error
 }
+
+// MeteredStorage collects read and write statistics.
+type MeteredStorage interface {
+	// Reads returns the cumulative number of read bytes of the underlying storage.
+	Reads() uint64
+	// Writes returns the cumulative number of written bytes of the underlying storage.
+	Writes() uint64
+}
+
+type meteredStorage struct {
+	s     Storage
```

Comment:
I think you can use inheritance and then just override relevant methods.

Reply:
Oh, I only saw the main comment, I'll change this too.